src/playlist-loader.js

```diff
@@ -268,11 +330,60 @@ export default class PlaylistLoader extends EventTarget {
         // only refresh the media playlist if no other activity is going on
         return;
       }
+      const media = this.media();
+
+      let uri = resolveUrl(this.master.uri, media.uri);
+
+      if (this.experimentalLLHLS && !media.endList) {
+        const query = [];
+
+        if (media.serverControl && media.serverControl.canBlockReload) {
+          const {preloadSegment} = media;
+          // next msn is a zero based value, length is not.
+          let nextMSN = media.mediaSequence + media.segments.length;
+
+          if (preloadSegment) {
+            const parts = preloadSegment.parts || [];
+            // _HLS_part is a zero based index
+            const nextPart = getKnownPartCount(media) - 1;
+
+            // if nextPart is > -1 and not equal to just the
+            // length of parts, then we know we had part preload hints
+            // and we need to add the _HLS_part= query
+            if (nextPart > -1 && nextPart !== (parts.length - 1)) {
+              // add existing parts to our preload hints
+              query.push(`_HLS_part=${nextPart}`);
+            }
+
+            // if we are requesting a nextPart or preload segment was added
+            // to our segment list. We are requesting a part of the preload segment
+            // or the full preload segment. Either way we need to go down by 1
+            // in nextMSN
```

Comment:
Is this because the `nextMSN` would be (and normally is, outside of this case) reflecting the predicted next media sequence number, rather than one that's reported in the playlist (the preload segment)?

Reply:
Yes its based on sequence number and has to take into account weather we our requesting something that was a preload segment, a part of a current segment, or the next segment after what we currently have.

Reply:
Maybe we can add those notes as a comment?